internal/pkg/aws/cloudformation/cloudformation.go

```diff
@@ -285,3 +308,78 @@ func (c *CloudFormation) deleteAndWait(in *cloudformation.DeleteStackInput) erro
 	}
 	return nil
 }
+
+// makeTagMatcher takes a set of wanted tags and returns a function which matches if the given set of
+// `cloudformation.Tag`s contains tags with all wanted keys and values
+func makeTagMatcher(wantedTags map[string]string) func([]*cloudformation.Tag) bool {
+	// Match all stacks if no desired tags are specified.
+	if len(wantedTags) == 0 {
+		return func([]*cloudformation.Tag) bool { return true }
+	}
+
+	return func(tags []*cloudformation.Tag) bool {
+		// Define a map to determine whether each wanted tag is a match.
+		tagsMatched := make(map[string]bool, len(wantedTags))
+
+		// We'll use this as a hash set to verify whether the stack's tags are present in the wanted tags list.
+		// We need two of these data structures because the empty tag value "" is shorthand for "just match the key."
+		hsWantedTags := make(map[string]bool)
+
+		// Populate the hash set and match map
+		for k := range wantedTags {
+			tagsMatched[k] = false
+			hsWantedTags[k] = true
+		}
+
+		// Loop over all tags on the stack and decide whether they match any of the wanted tags.
+		for _, tag := range tags {
+			tagKey := aws.StringValue(tag.Key)
+			tagValue := aws.StringValue(tag.Value)
+			/*
+				if  hsWantedTags[tagKey] &&           checks whether the current stack tag is one of the wanted tags.
+				    (wantedTags[tagKey] == tagValue   checks whether the current stack tag's value matches the wanted value
+				    || wantedTags[tagKey] == "") {    checks whether the wanted tag is empty (match all tags with this key, don't worry about value)
+			*/
+			if hsWantedTags[tagKey] && (wantedTags[tagKey] == tagValue || wantedTags[tagKey] == "") {
```

Comment:
Do we need `hsWantedTags`? Isn't `hsWantedTags[tagKey]` equivalent to `wantedTags[tagKey] == tagValue || wantedTags[tagKey] == ""`?